frontend/packages/ui/src/components/modal/modal.tsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useRef, MouseEvent, useCallback } from 'react';
```

Comment:
`Modal` already exists in a `@gear-js/vara-ui`. We should extend and improve UI components from library, so let's not duplicate them.